tests/run-make/incr-prev-body-beyond-eof/rmake.rs

```diff
@@ -0,0 +1,28 @@
+// After modifying the span of a function, if the length of
+// the span remained the same but the end line number became different,
+// this would cause an internal compiler error (ICE), fixed in #76256.
+
+// This test compiles main.rs twice, first with end line 16 and
+// then with end line 12. If compilation is successful, the end line
+// was hashed by rustc in addition to the span length, and the fix still
+// works.
+
+use run_make_support::{rustc, target, tmp_dir};
+use std::fs;
+
+fn main() {
+    fs::create_dir(tmp_dir().join("src"));
+    fs::create_dir(tmp_dir().join("incr"));
+    fs::copy("a.rs", tmp_dir().join("main.rs"));
+    rustc()
+        .incremental(tmp_dir().join("incr"))
+        .input(tmp_dir().join("src/main.rs"))
+        .target(target())
+        .run();
+    fs::copy("b.rs", tmp_dir().join("main.rs"));
+    rustc()
+        .incremental(tmp_dir().join("incr"))
```

Comment:
The `tmp_dir().join(...)` pattern seems to be repeating quite often. It could be worth it to introduce something like `fn tmp_path(path: &str)` to simplify it, it's a bit hard to read.

Reply:
I think we should add a helper, but in a separate PR (unrelated changes), I think it's fine for this PR to keep the join for now, I'll send another PR, so this PR can either rebase on top of that or I'll update the other PR to cover join in this PR too.